src/components/Board.jsx

```diff
@@ -22,6 +29,51 @@ const Board = ({ id }) => {
     const columnOrderArray = board.columnOrder
     const { columns } = board
 
+    // TODO, move this function into utils folder
+    const onDragEnd = async (result) => {
+        const { destination, source, draggableId } = result
+
+        if (!destination) {
+            return
+        }
+
+        if (destination.droppableId === source.droppableId && destination.index === source.index) {
+            return
+        }
+
+        // When task is moved within one column
+        if (destination.droppableId === source.droppableId) {
+            const column = columns.find((col) => col.id === source.droppableId)
+            const newTaskOrder = Array.from(column.taskOrder)
+
+            newTaskOrder.splice(source.index, 1)
+            newTaskOrder.splice(destination.index, 0, draggableId)
```

Comment:
splice mutates the array, suggest using a new array variable for mutated data.

Reply:
Don't see a problem here since a new array is created from the old task order array before we splice anything. So we are not touching the original data used in rendering the components